### PR TITLE
Bluetooth: tester: Add logging in core and GAP

### PR DESCRIPTION
**Pull request by carlescufi**
See commit messages.

### CODE REVIEW

File: tests/bluetooth/tester/src/gap.c [hunk 1]
```diff
@@ -701,13 +723,17 @@ void tester_handle_gap(u8_t opcode, u8_t index, u8_t *data,
...
+			LOG_WRN("index != BTP_INDEX_NONE: opcode: 0x%x"
+				 "index: 0x%x", opcode, index);
 			return;
 		}
 		break;
 	default:
 		if (index != CONTROLLER_INDEX){
 			tester_rsp(BTP_SERVICE_ID_GAP, opcode, index,
 				   BTP_STATUS_FAILED);
+			LOG_WRN("index != BTP_INDEX_NONE: opcode: 0x%x"
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: index != CONTROLLER_INDEX

File: tests/bluetooth/tester/src/bttester.c [hunk 2]
```diff
@@ -104,6 +104,7 @@ static void register_service(u8_t *data, u16_t len)
 		break;
 #endif /* CONFIG_BT_MESH */
 	default:
+		LOG_WRN("unknown id: 0x%x", cmd->id);
```
Comment (joerchan) [inline tests/bluetooth/tester/src/bttester.c]: would it make sense to print it as 0x%02x?

File: tests/bluetooth/tester/src/gap.c [hunk 3]
```diff
@@ -295,6 +296,7 @@ static void stop_advertising(const u8_t *data, u16_t len)
 	if (bt_le_adv_stop() < 0) {
 		tester_rsp(BTP_SERVICE_ID_GAP, GAP_STOP_ADVERTISING,
 			   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		LOG_ERR("Failed to stop advertising");
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: It would be helpful to get the error code as well.

File: tests/bluetooth/tester/src/gap.c [hunk 4]
```diff
@@ -462,6 +466,7 @@ static void stop_discovery(const u8_t *data, u16_t len)
 	u8_t status = BTP_STATUS_SUCCESS;
 
 	if (bt_le_scan_stop() < 0) {
+		LOG_ERR("Failed to stop scanning");
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: error code here as well.

File: tests/bluetooth/tester/src/gap.c [hunk 5]
```diff
@@ -496,10 +502,12 @@ static void disconnect(const u8_t *data, u16_t len)
 	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, (bt_addr_le_t *)data);
 	if (!conn) {
 		status = BTP_STATUS_FAILED;
+		LOG_ERR("Unknown device");
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: switch device with connection? (more than one place).

File: tests/bluetooth/tester/src/gap.c [hunk 6]
```diff
@@ -593,11 +602,13 @@ static void pair(const u8_t *data, u16_t len)
 
 	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, (bt_addr_le_t *)data);
 	if (!conn) {
+		LOG_ERR("Unknown device");
 		status = BTP_STATUS_FAILED;
 		goto rsp;
 	}
 
 	if (bt_conn_set_security(conn, BT_SECURITY_L2)) {
+		LOG_ERR("Failed to set security");
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: Get the error code here as well.

File: tests/bluetooth/tester/src/gap.c [hunk 7]
```diff
@@ -795,6 +823,7 @@ static void tester_init_gap_cb(int err)
 u8_t tester_init_gap(void)
 {
 	if (bt_enable(tester_init_gap_cb) < 0) {
+		LOG_ERR("Unable to enable Bluetooth");
```
Comment (joerchan) [inline tests/bluetooth/tester/src/gap.c]: error code here.

**Review by joerchan — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
@joerchan thanks for the review, comments addressed

**Review by joerchan — APPROVED** [review]
Logs look good. But RTT does not  work for me with the tester for some reason.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**